### PR TITLE
Support `StatePrep` mid-circuit for `DefaultQubit2`

### PR DESCRIPTION
**Pull request by Jaybsoni**
**Context:**
Working towards the goal of support LCUs in Pennylane. This PR allows StatePrep operations to be called in the middle of a circuit without raising errors.

**Description of the Change:**
- Removed an error the used to be raised if more than 1 `StatePrep` operation was present in the circuit. 
- The `expand_fn()` in `device/qubit/preprocess.py` is updated to allow the first operation in the circuit to remain un-expanded if it is a `StatePrep` operation, otherwise expanding all other instances of `StatePrep`

### CODE REVIEW

**Comment by Jaybsoni** [comment]
[sc - 42794]

**Comment by Jaybsoni** [comment]
[sc-42794]

File: pennylane/devices/qubit/preprocess.py [hunk 1]
```diff
@@ -248,14 +248,18 @@ def expand_fn(circuit: qml.tape.QuantumScript) -> qml.tape.QuantumScript:
...
-
-    if not all(_accepted_operator(op) for op in circuit._ops):
+    if not all(_accepted_operator(op) for op in circuit.operations):
         try:
+            prep_op = circuit.operations[0]
+            new_prep = (
+                [prep_op]
+                if isinstance(prep_op, StatePrep)
+                else list(_operator_decomposition_gen(prep_op, _accepted_operator))
+            )
```
[the remark this answers is not in the document]
Comment (Jaybsoni) [reply pennylane/devices/qubit/preprocess.py:258]: That's much cleaner 💯

File: tests/devices/qubit/test_preprocess.py [hunk 2]
```diff
@@ -260,6 +252,33 @@ def test_expand_fn_non_commuting_measurements(self):
...
+            qml.QubitStateVector([0, 1], wires=1),
+            qml.BasisState([1], wires=0),
+            qml.RZ(0.123, wires=1),
+        ]
+        measurements = [qml.expval(qml.PauliZ(0)), qml.probs()]
+        tape = QuantumScript(ops=ops, measurements=measurements)
+
+        expanded_tape = expand_fn(tape)
+        expected = [
+            prep_op,
```
Comment (soranjh) [inline tests/devices/qubit/test_preprocess.py:272]: Why it is not decomposed if it is the first operation?

Comment (Jaybsoni) [reply tests/devices/qubit/test_preprocess.py:272]: If it's the first operation, we just update the state vector of the simulator itself

**Review by soranjh — APPROVED** [review]

**Review by albi3ro — APPROVED** [review]
👍 looks good ;)

**Pull request merged**